js/src/relay/WalletLinkRelay.ts

```diff
@@ -352,6 +353,20 @@ export class WalletLinkRelay {
     })
   }
 
+  public static makeRequestId(): number {
+    // max nextId == max int32 for compatibility with mobile
+    this._nextRequestId = (this._nextRequestId + 1) % 0x7fffffff
+    const id = this._nextRequestId
+    const idStr = prepend0x(id.toString(16))
+    // unlikely that this will ever be an issue, but just to be safe
+    const callback = WalletLinkRelay.callbacks.get(idStr)
+    if (callback) {
+      // TODO - how to handle this case
```

Comment:
Wouldn't it be safer to just try generating a new `id`?

If I understood the docs correctly, the id is for keeping track of requests and responses when sent over a websocket, so the consequence here would be mismatching a response to a request. No?

Reply:
This code is taken from https://github.com/CoinbaseWallet/cipher-web-view/blob/50dd82456575dccda41f1f4629372096b6969d92/src/CipherProvider.ts#L133

ID is sequential, and once it reaches max int32, which will probably never happen, it will start from ID 0 again. When that happens, it's probably safe to say that the callback for ID 0 has already been called. (since 2 billion+ other requests have been called since then)